main.tf

```diff
@@ -1,6 +1,6 @@
 # Create rgs as defined by var.virtual_wan
 resource "azurerm_resource_group" "rg" {
-  for_each = var.resource_group_name != null ? toset([var.resource_group_name]) : toset([])
+  for_each = var.create_rg ? toset([var.resource_group_name]) : toset([])
 
   location = var.location
   name     = var.resource_group_name
```

Comment:
This should use the `azurerm_resource_group.rg.name` to build an implicit dependency instead. No need for the explicit dependency.